### ISSUE TITLE
How to get all errors from all input? 

### ISSUE THREAD

**Issue by renttless** [Respect/Validation#649]
Hey there. I recently started using this package and I find it very particular and nice. I've being through the docs but they seem confusing (to me), that's why I came here for enlightenment.

I want to get all the errors from all the input by key. I'm currently doing like this:

``` php
try {
    v::key('activity', v::notEmpty())->key('department', v::notEmpty())
        ->assert(['activity' => '', 'department' => '']);
} catch(NestedValidationException $e) {
    var_dump($e->getErrors('activity', 'department'));
}
```

And this is the output I get:

``` php
[
    'activity' => 'Key activity must be valid',
    'department' => '',
]
```

If I use `$e->findMessages(['notEmpty']);` This is what I get:

``` php
[
    'notEmpty' => 'activity must not be empty',
]
```

So, I don't get how to get it working. This is what I want to achieve:

``` php
[
    'activity' => 'activity can\'t be empty',
    'department' => 'can\'t be empty',
]
```

Or better yet, an array with all the errors that failed for each field, for example:

``` php
[
  'activity' => [
        'activity can\'t be empty',
        'activity must be a string',
        // and like this with all the properties that failed.
    ],
]
```

Thanks in advance for your help!

**Comment by mscreenie** [comment]
+1 Trying to do the same. So far I've only had success using Respect validating a basic contact form.

Apart from that its incredibly difficult using it in large scale projects without hard coding a lot of things.

**Comment by henriquemoody** [comment]
Fixed by #957.

I've changed how `getMessages()`, check out [get_messages.phpt](https://github.com/Respect/Validation/blob/master/tests/integration/get_messages.phpt) and [get_messages_with_replacements.phpt](https://github.com/Respect/Validation/blob/master/tests/integration/get_messages_with_replacements.phpt) to see how it works.

It will be released with the version 2.0. If any of you want to speed things up take a look at #921!

Thanks for reporting! :panda_face:
